x-pack/test/functional_enterprise_search/page_objects/app_search.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+import { TestSubjects } from '../../../../../test/functional/services/test_subjects';
+
+export function AppSearchPageProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+  const testSubjects = getService('testSubjects') as TestSubjects;
+
+  return {
+    async navigateToPage() {
+      return await PageObjects.common.navigateToApp('app_search');
+    },
+
+    async getEngineLinks() {
+      const engines = await testSubjects.find('appSearchEngines');
+      return await testSubjects.findAllDescendant('engineNameLink', engines);
+    },
```

Comment:
So, I've kinda had this conversation before but for Cypress, but I get mildly concerned with (what I think is?) over-DRYing stuff out instead of just using the native APIs.

1. It requires someone new to the testing framework API to jump between 2 different files to understand what the helper is actually doing
2. If we continue to use a `getSomeSelector()` helper for every single new test we add, this file is going to quickly balloon in size and become increasingly difficult to parse/read

This isn't a change request but more a general philosophy that but I'd prefer to keep selector helpers in the file they're used in, and only start DRYing them out to a shared file once more than 2-3+ tests duplicate them.

Reply:
That's a fair point, thanks.